src/main/java/com/marafone/marafone/game/active/ActiveGameServiceImpl.java

```diff
@@ -38,16 +48,22 @@ public Long createGame(CreateGameRequest createGameRequest, User user) {
                 .createdAt(LocalDateTime.now())
                 .playersList(new ArrayList<>(List.of(gamePlayer)))
                 .rounds(new LinkedList<>())
+                .name(createGameRequest.getGameName())
                 .gameType(createGameRequest.getGameType())
                 .owner(user)
                 .joinGameCode(createGameRequest.getJoinGameCode())
                 .build();
 
-        Long id = activeGameRepository.put(game);
+        return activeGameRepository.put(game);
+    }
 
-        game.setName(String.valueOf(id));
+    // checks if game with arg name already exists
+    // among games waiting for players to join
 
-        return id;
+    public boolean doesNotStartedGameAlreadyExist(String name) {
+        return getPublicGames().stream()
```

Comment:
You can't really use getPublicGames() here because getPublicGames will be cached using proxy design pattern (#2), which means it might not have the newest information. activeGameRepositoryCache will ask activeGameRepository for newest information only if the cached value was created more than 2 seconds ago.